### PR TITLE
Added cups 2.3.3

### PR DESCRIPTION
**Pull request by Janrupf**
This pull request adds the Common UNIX Printing System. The package compiles nearly out of the box, only removing a ``pthread_cancel`` is required to get it to work. Since CUPS does not rely on this function to be present, it does not impact the functionality.

### CODE REVIEW

File: packages/cups/build.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+TERMUX_PKG_VERSION="2.3.3"
+TERMUX_PKG_SRCURL="https://github.com/apple/cups/releases/download/v${TERMUX_PKG_VERSION}/cups-${TERMUX_PKG_VERSION}-source.tar.gz"
+TERMUX_PKG_SHA256="261fd948bce8647b6d5cb2a1784f0c24cc52b5c4e827b71d726020bcc502f3ee"
+TERMUX_PKG_DEPENDS="libiconv, libcrypt"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_CONFFILES="etc/cups/cups-files.conf, etc/cups/cupsd.conf, etc/cups/snmp.conf"
+
+termux_step_make() {
+	cd "$TERMUX_PKG_BUILDDIR"
+	make LIBS="-lz -pthread -lm -lcrypt -liconv -lz -L$TERMUX_PREFIX/lib -Wl,-rpath=$TERMUX_PREFIX/lib -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,relro,-z,now -Wl,-rpath,$TERMUX_PREFIX/lib" -j$(nproc)
```
Comment (landfillbaby) [inline packages/cups/build.sh:14]: duplicate `-lz` and `-Wl,rpath..`

Comment (xeffyr) [reply packages/cups/build.sh:14]: And `-j$(nproc)` should be replaced with `-j${TERMUX_MAKE_PROCESSES}`.

File: packages/cups/build.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+termux_step_make_install() {
+	cd $TERMUX_PKG_BUILDDIR
+	make install DESTDIR="$TERMUX_PKG_MASSAGEDIR"
```
Comment (xeffyr) [inline packages/cups/build.sh:19]: Installation into $TERMUX_PKG_MASSAGEDIR should never be done. Instead everything **must** be installed into $TERMUX_PREFIX.

File: packages/cups/build.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+termux_step_post_massage() {
+	echo "Correcting packaging..."
+	cp -a "$TERMUX_PKG_MASSAGEDIR/etc"/* "$TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/etc"
+	cp -a "$TERMUX_PKG_MASSAGEDIR/usr"/* "$TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX"
+	rm -rf "$TERMUX_PKG_MASSAGEDIR/etc"
+	rm -rf "$TERMUX_PKG_MASSAGEDIR/usr"
```
Comment (xeffyr) [inline packages/cups/build.sh:27]: Remove this "hack" and fix the package properly. It was built with **wrong prefix** and will search conffiles in /etc.

File: packages/cups/thread.patch [hunk 4]
```diff
@@ -0,0 +1,13 @@
+--- src/cups/thread.c	2020-07-04 23:37:13.920000000 +0000
++++ src-mod/cups/thread.c	2020-07-04 23:38:15.290000000 +0000
+@@ -153,7 +153,9 @@
+ void
+ _cupsThreadCancel(_cups_thread_t thread)/* I - Thread ID */
+ {
+-  pthread_cancel(thread);
```
Comment (xeffyr) [inline packages/cups/thread.patch:7]: You just broke threading.

**Comment by Janrupf** [comment]
I'm not sure how to do packaging. It seems that the Makefile ignores PREFIX and tries to install into system directories.

**Comment by xeffyr** [comment]
> It seems that the Makefile ignores PREFIX and tries to install into system directories.

It ignores it only for init.d scripts which are useless in Termux anyway. You need to disable their installation, probably by patching makefile if there no option to disable them.

**Pull request merged**